### DISCUSSION TITLE
how to get cookies which are setted in server only using route handlers

### DISCUSSION THREAD

**Discussion by Arunjenson** [vercel/next.js#61779] [App Router]
for server side cookie setting i have created a route.js file inside products/api 
because i need the value of cookie on the /products/page.jsx in server side only 
but how to get the cookie value in the page.jsx here?

**Comment by khuezy** [comment]
https://nextjs.org/docs/app/api-reference/functions/cookies#cookiesgetname

**Reply by Arunjenson** [reply]
ya im able to set the cookie in the route handler products/api, so but i need the server side setted cookie in my products /page.jsx how to get the cookie in page.jsx?

**Reply by khuezy** [reply]
I'm confused, the docs I link literally has an example that lets you grab the cookies in the `page.ts(x)` file

**Reply by Arunjenson** [reply]
let me make it clear , yes i m able to set the cookie via route handler in server and send cookie header as a response , so cookie will set only if we have a request to that particular route , but when i change the url lets say /product , i cant see the setted cookie in the browser its getting cleared away , since its getting cleared not able to access the cookie using cookies.get(),
also instead of first manually having a request to cookie route handler can we direclty to /product then first it should send request to the route handler to set the cookie and set the cookie and /product page loads with the server side cookie
if your free can you add can example here? @khuezy

**Comment by Arunjenson** [comment]
https://github.com/vercel/next.js/issues/52799
